src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -324,7 +372,14 @@ async Task ExecuteInvocation()
                             object result;
                             try
                             {
-                                result = await ExecuteHubMethod(methodExecutor, hub, arguments);
+                                bool methodInvoked;
+                                (methodInvoked, result) = await ExecuteHubMethod(methodExecutor, hub, arguments, connection, scope.ServiceProvider);
+                                if (!methodInvoked)
+                                {
+                                    await SendInvocationError(hubMethodInvocationMessage.InvocationId, connection,
```

Comment:
This seems overly defensive to me.

Reply:
@BrennanConroy After re-thinking this can we just support a way to return an invocation error instead of inferring anything? We can treat it like `HubException`? We support a way that you can opt into sending a custom InvocationError but we don't do anything else.

Reply:
> can we just support a way to return an invocation error

That's what the `HubResult` is for

Reply:
Undo